src/pre-engagement-form/localization.tsx

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (C) 2021-2023 Technology Matters
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see https://www.gnu.org/licenses/.
+ */
+
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import type { Manager } from '@twilio/flex-webchat-ui';
+import PubSub from 'pubsub-js';
+
+type LocalizationProviderProps = {
+  manager: Manager;
+  children: JSX.Element;
+};
+
+const LocalizationContext = createContext({});
+const STRINGS_UPDATED = 'STRINGS_UPDATED';
+
+export const notifyStringsUpdated = () => PubSub.publish(STRINGS_UPDATED);
+
+export const LocalizationProvider: React.FC<LocalizationProviderProps> = ({ manager, children }) => {
```

Comment:
Sorry if this is a silly question, but have tried passing down `strings` as a prop here rather than `manager`? That object reference should change when the translation is updated, and trigger a re-render of the context provider (and all of it's descendants). If so, maybe there's no need for the `pubsub` addition here?

Again sorry if this is nonsense, maybe there's something I'm missing that justifies the current approach!

Reply:
Yeah, but there were some issues.
First, this is how this component is added:
```
FlexWebChat.PreEngagementCanvas.Content.replace(<PreEngagementForm key="pre-engagement" manager={manager} />);
```
It's not under "React", per si, so no re-rendering would be triggered.
Then, another option would be to get `strings` from WebChat redux store, but WebChat does not store them on redux 🥲.

What we could do in the future is to manage `strings` in our own redux state (create state/actions/reducer for the strings). We'll just need to be careful on keeping `manager.strings` and redux `strings` in sync.

Reply:
@stephenhand This is literally what I was just replying to Gian 😅